packages/synapse-react-client/src/utils/functions/TableColumnSchemaUtils.ts

```diff
@@ -47,23 +74,34 @@ export async function createTableUpdateTransactionRequest(
     .list
 
   // now that all columns have been created, figure out the column changes (create, update, and no-op)
-  // keep track of column ids to figure out what columns were deleted.
   const changes: ColumnChange[] = []
-  const columnIds: Set<string | null> = new Set()
+  const columnIdsThatWereChangedOrAdded: Set<string> = new Set()
   for (let i = 0; i < proposedSchema.length; i++) {
-    const oldColumnId: string | null = proposedSchema[i].id ?? null
-    const newColumnId: string | null = createdColumnModels[i].id ?? null
-    columnIds.add(oldColumnId)
-    columnIds.add(newColumnId)
-    if (oldColumnId !== newColumnId) {
-      changes.push({ oldColumnId, newColumnId })
+    const columnIdInProposedSchema: string | null = proposedSchema[i].id ?? null
+    const newColumnId: string = createdColumnModels[i].id
+    if (columnIdInProposedSchema)
+      columnIdsThatWereChangedOrAdded.add(columnIdInProposedSchema)
+    columnIdsThatWereChangedOrAdded.add(newColumnId)
+    if (
+      // The column has changed
+      columnIdInProposedSchema != null &&
+      columnIdInProposedSchema !== newColumnId
+    ) {
+      changes.push({ oldColumnId: columnIdInProposedSchema, newColumnId })
+    } else if (
+      // the column is new and is defined by the user
+      columnIdInProposedSchema == null ||
+      // the column is new but already has an ID (e.g. we are adding a default column to a view)
+      !oldColumnModelId2Model.has(columnIdInProposedSchema)
```

Comment:
This should never happen now; IDs should be removed from _any_ new column before this method is called, including default columns